easybuild/framework/easyblock.py

```diff
@@ -846,9 +845,12 @@ def make_module_extra(self):
 
         # EBROOT + EBVERSION + EBDEVEL
         environment_name = convert_name(self.name, upper=True)
-        txt += self.module_generator.set_environment(ROOT_ENV_VAR_NAME_PREFIX + environment_name, "$root")
+        #todo this is only valid for Lua now
+        # This is a bit different in Lua due to string quoting rules in Lua and in Tcl - so $root cannot be used easily.
+        # so we resort to rendering our internal variables and quote them in the set_environment() like all other values.
+        txt += self.module_generator.set_environment(ROOT_ENV_VAR_NAME_PREFIX + environment_name, self.installdir)
         txt += self.module_generator.set_environment(VERSION_ENV_VAR_NAME_PREFIX + environment_name, self.version)
-        devel_path = os.path.join("$root", log_path(), ActiveMNS().det_devel_module_filename(self.cfg))
+        devel_path = os.path.join(self.installdir, log_path(), ActiveMNS().det_devel_module_filename(self.cfg))
```

Comment:
either fix the lua thingie you mentioned above, or remove the comment altogether (and pretend we never expect to read the files)